mobly/controllers/android_device.py

```diff
@@ -372,6 +373,29 @@ def __init__(self, serial=''):
         # names, values are the clients: {<attr name string>: <client object>}.
         self._snippet_clients = {}
 
+
+    @contextlib.contextmanager
+    def capture_logcat(self, fd=None):
+        """Captures logcat output produced during this context
+        manager's lifetime to given file object.
+
+        Args:
+            fd: file (or compatible) object, e.g. returned by open().
```

Comment:
This allow you to send in a tempfile object and have it write to that. Or, you can open an actual file and manage it yourself.